### PR TITLE
bfconvert: fix the writer's XY coordinate when the input image is cropped

### LINKED ISSUE #3461: bfconvert and showinf 6.3.0 issues
1) bvconvert issue
D:\vsi>bfconvert -series 7 -tiley 512 -tiley 512 -compression JPEG -crop 32256,8704,5120,5120 image.vsi converted.tif
image.vsi
CellSensReader initializing image.vsi
[CellSens VSI] -> converted.tif [Tagged Image File Format]
Tile size = 512 x 512
Exception in thread "main" loci.formats.FormatException: X:32256 must be < 5120
        at loci.formats.FormatWriter.checkParams(FormatWriter.java:453)
        at loci.formats.out.TiffWriter.saveBytes(TiffWriter.java:220)
        at loci.formats.tools.ImageConverter.convertTilePlane(ImageConverter.java:878)
        at loci.formats.tools.ImageConverter.convertPlane(ImageConverter.java:763)
        at loci.formats.tools.ImageConverter.testConvert(ImageConverter.java:691)
        at loci.formats.tools.ImageConverter.main(ImageConverter.java:1051)

The image in that series is 49512x38578.
The command line option "-crop" says:
 -crop: crop images before converting; argument is 'x,y,w,h'

2) showinf issue
The command line option "-series" says:
-series: specify which image series to read

If I leave off -series on say my above vsi image, all 16 series info shows.  If I put a -series 7, 16 series will be displayed.  Shouldn't it only show the info for series 7?

### PR DESCRIPTION
**Pull request by melissalinkert**
Fixes #3461.  The previous behavior was to use the same XY coordinate when reading and writing a tile.  This is fine when there is no cropping, but otherwise results in a coordinate that fails the writer's bounds check.

Without this PR, ``` bfconvert -crop 32256,8704,5120,5120 "test&sizeX=49512&sizeY=38578.fake" test.tif``` should throw:

```
Exception in thread "main" loci.formats.FormatException: X:32256 must be < 5120
	at loci.formats.FormatWriter.checkParams(FormatWriter.java:453)
	at loci.formats.out.TiffWriter.saveBytes(TiffWriter.java:220)
	at loci.formats.tools.ImageConverter.convertTilePlane(ImageConverter.java:878)
	at loci.formats.tools.ImageConverter.convertPlane(ImageConverter.java:763)
	at loci.formats.tools.ImageConverter.testConvert(ImageConverter.java:691)
	at loci.formats.tools.ImageConverter.main(ImageConverter.java:1051)
```

With this PR, the same test should successfully write a 5120x5120 file without throwing an exception.  The resulting file should be read correctly as well.

### CODE REVIEW

**Comment by sbesson** [comment]
:+1: sbesson@1b3199e129cc91fb4b0023560d4ebedb42f49ff7 contains a minimal unit test reproducing the original failure and which should pass with this fix included.

**Comment by melissalinkert** [comment]
Thanks for the test; commit cherry-picked onto this PR's branch as 0913d76.

**Comment by mtbc** [comment]
Just beforehand there's the conditional,
```java
          if (nTileRows > 1) {
            tileY = 0;
          }
          if (nTileCols > 1) {
            tileX = 0;
          }
        }
```
It's okay that we're unzeroing them again for the tile writing?

File: components/bio-formats-tools/test/loci/formats/tools/ImageConverterTest.java [hunk 1]
```diff
@@ -216,4 +218,20 @@ public void testSPWSeries() throws FormatException, IOException {
     };
     assertConversion(args);
   }
+
+  @Test
+  public void testCrop() throws FormatException, IOException {
+    outFile = tempDir.resolve("test.ome.tiff").toFile();
+    String[] args = {
+      "-tilex", "128", "-tiley", "128",
+      "-crop", "256,256,256,256", "test.fake", outFile.getAbsolutePath()};
```
Comment (mtbc) [inline components/bio-formats-tools/test/loci/formats/tools/ImageConverterTest.java]: Maybe could also include a test that isn't exactly on tile boundaries.

**Comment by mtbc** [comment]
Despite comments (which arise mostly from my ignorance), actually looks great and works fine as far as I can tell, no objections here. :+1:

**Comment by melissalinkert** [comment]
> Just beforehand there's the conditional,
> 
> ```java
>           if (nTileRows > 1) {
>             tileY = 0;
>           }
>           if (nTileCols > 1) {
>             tileX = 0;
>           }
>         }
> ```
> 
> It's okay that we're unzeroing them again for the tile writing?

That block (https://github.com/ome/bioformats/pull/3463/files#diff-3ab3923a0dab584a4eadb99362e60687R857) only occurs if this condition is met:

https://github.com/ome/bioformats/pull/3463/files#diff-3ab3923a0dab584a4eadb99362e60687R831

which should only happen if we're writing tiles to separate files, which isn't the case here.  Admittedly, I didn't test the separate files case and I don't think we have tests for it either - maybe warrants a follow-up issue?

File: components/bio-formats-tools/test/loci/formats/tools/ImageConverterTest.java (hunk 1 above)
Comment (melissalinkert) [reply components/bio-formats-tools/test/loci/formats/tools/ImageConverterTest.java]: Added in 7a11de2

**Comment by mtbc** [comment]
Certainly happy for the question to be relegated to a follow-up issue, it'll be good to get this present fix in.

File: components/bio-formats-tools/test/loci/formats/tools/ImageConverterTest.java (hunk 1 above)
Comment (mtbc) [reply components/bio-formats-tools/test/loci/formats/tools/ImageConverterTest.java]: :+1:

**Comment by dgault** [comment]
Merging this initial fix as it fixes the initial issue and the unit test looks good. Further tests and fixes will be added in follow up PR's
